### PR TITLE
Remove unnecessary auth checks

### PR DESCRIPTION
**Pull request by nwalters512**
These are all covered by the middleware that runs for the entire `/pl/course_instance/:course_instance_id/instance_question/:instance_question_id` route prefix.

https://github.com/PrairieLearn/PrairieLearn/blob/ed27bcfa732e8a8e1ff6833a06499a34fa85d7bf/server.js#L1335-L1341

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1369,42 +1366,26 @@ module.exports.initExpress = function () {
   );
   app.use(
     '/pl/course_instance/:course_instance_id/instance_question/:instance_question_id/clientFilesQuestion',
-    [
-      require('./middlewares/selectAndAuthzInstanceQuestion'),
-      require('./middlewares/studentAssessmentAccess'),
-      require('./pages/clientFilesQuestion/clientFilesQuestion'),
-    ]
+    [require('./pages/clientFilesQuestion/clientFilesQuestion')]
```
Comment (mwest1066) [inline server.js:1369]: We can get rid of the `[]` here if we want (and similarly elsewhere).
```suggestion
    require('./pages/clientFilesQuestion/clientFilesQuestion')
```

File: server.js [hunk 2]
```diff
@@ -946,10 +946,7 @@ module.exports.initExpress = function () {
   // Submission files
   app.use(
     '/pl/course_instance/:course_instance_id/instructor/instance_question/:instance_question_id/submission/:submission_id/file',
-    [
-      require('./middlewares/selectAndAuthzInstanceQuestion'),
```
Comment (mwest1066) [inline server.js:950]: I think this one is actually needed?

Comment (nwalters512) [reply server.js]: Good catch! Resolved in 2aaec9c.

File: server.js (hunk 1 above)
Comment (nwalters512) [reply server.js]: Done in 2aaec9c.

Comment (mwest1066) [reply server.js]: Not that it really matters, but it looks like you missed the ones for `legacyQuestionFile` and `legacyQuestionText`.

**Review by mwest1066 — APPROVED** [review]
Looks good. See https://github.com/PrairieLearn/PrairieLearn/pull/7377#discussion_r1152648245 for a minor tweak if you want to do it.

File: server.js (hunk 1 above)
Comment (nwalters512) [reply server.js]: Whoops, indeed - addressed in 5e59a34.

**Pull request merged**